go-controller/pkg/ovn/kubevirt.go

```diff
@@ -0,0 +1,232 @@
+package ovn
+
+import (
+	"fmt"
+	"net"
+
+	corev1 "k8s.io/api/core/v1"
+	kapi "k8s.io/api/core/v1"
+	utilnet "k8s.io/utils/net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kubevirt"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+func (oc *DefaultNetworkController) ensureDHCPOptionsForVM(pod *corev1.Pod, lsp *nbdb.LogicalSwitchPort) error {
+	if !kubevirt.AllowPodBridgeNetworkLiveMigration(pod.Annotations) {
+		return nil
+	}
+
+	switchNames, err := oc.getSwitchNames(pod)
+	if err != nil {
+		return fmt.Errorf("failed configuring dhcp when getting switch current and original name: %v", err)
+	}
+	var switchSubnets []*net.IPNet
+	if switchSubnets = oc.lsManager.GetSwitchSubnets(switchNames.Original); switchSubnets == nil {
+		return fmt.Errorf("subnet not found for switch %s to configuare DHCP at lsp %s", switchNames.Original, lsp.Name)
+	}
+	// Fake router to delegate on proxy arp mechanism
+	vmName, ok := pod.Labels[kubevirt.VMLabel]
+	if !ok {
+		return fmt.Errorf("missing %s label at pod %s/%s when configuaring DHCP", kubevirt.VMLabel, pod.Namespace, pod.Name)
+	}
+	dhcpConfig, err := kubevirt.ComposeDHCPConfig(oc.watchFactory, vmName, switchSubnets)
+	if err != nil {
+		return fmt.Errorf("failed composing DHCP options: %v", err)
+	}
+	var (
+		dhcpv4OptionsPredicate, dhcpv6OptionsPredicate func(*nbdb.DHCPOptions) bool
+	)
+	if dhcpConfig.V4Options != nil {
+		dhcpv4OptionsDbObjectID := libovsdbops.NewDbObjectIDs(libovsdbops.VirtualMachineDHCPOptions, "DefaultController",
+			map[libovsdbops.ExternalIDKey]string{
+				libovsdbops.ObjectNameKey: dhcpConfig.V4Options.Cidr,
+				libovsdbops.HostnameIndex: vmName,
+			})
+		dhcpConfig.V4Options.ExternalIDs = dhcpv4OptionsDbObjectID.GetExternalIDs()
+		dhcpConfig.V4Options.ExternalIDs[kubevirt.VMLabel] = vmName
+		dhcpConfig.V4Options.ExternalIDs["namespace"] = pod.Namespace
+		dhcpv4OptionsPredicate = libovsdbops.GetPredicate[*nbdb.DHCPOptions](dhcpv4OptionsDbObjectID, nil)
+	}
+	if dhcpConfig.V6Options != nil {
+		dhcpv6OptionsDbObjectID := libovsdbops.NewDbObjectIDs(libovsdbops.VirtualMachineDHCPOptions, "DefaultController",
+			map[libovsdbops.ExternalIDKey]string{
+				libovsdbops.ObjectNameKey: dhcpConfig.V6Options.Cidr,
+				libovsdbops.HostnameIndex: vmName,
+			})
+		dhcpConfig.V6Options.ExternalIDs = dhcpv6OptionsDbObjectID.GetExternalIDs()
+		dhcpConfig.V6Options.ExternalIDs[kubevirt.VMLabel] = vmName
+		dhcpConfig.V6Options.ExternalIDs["namespace"] = pod.Namespace
```

Comment:
Just want to make sure we have consciously decided against using the `k8s.ovn.org` prefix here.

Reply:
Adding it, since we are in control of it.